test/unit/should_generate_field_resolver_interfaces/expected.kt

```diff
@@ -18,6 +18,7 @@ interface QueryCompletableFuture {
     fun nonNullableResolver(arg: InputTypeGenerateFieldResolverInterfaces): java.util.concurrent.CompletableFuture<String>
 }
 
+@GraphQLValidObjectLocations(locations = [GraphQLValidObjectLocations.Locations.INPUT_OBJECT])
```

Comment:
why does this need this annotation INPUT_OBJECT. This doesn't end with `Input` and also this is an interface.

Reply:
This class is generated from [this type](https://github.com/ExpediaGroup/graphql-kotlin-codegen/blob/main/test/unit/should_generate_field_resolver_interfaces/schema.graphql#L8)

Reply:
Basically any `input` type that doesn't have a matching `type` is going to get this annotation

Reply:
And vice versa - any `type` that doesn't have a matching `input` will get `@GraphQLValidObjectLocations(locations = [GraphQLValidObjectLocations.Locations.OBJECT])`

Reply:
Will add detailed docs on this later